src/lib/twitter_web/controllers/post_controller.ex

```diff
@@ -0,0 +1,68 @@
+defmodule TwitterWeb.PostController do
+  @moduledoc """
+  The Post Controller.
+  """
+  use TwitterWeb, :controller
+
+  alias Twitter.Tweets
+
+  @doc """
+  Creates a post and returns the homepage of the application
+  """
+  @spec create(Plug.Conn.t(), map() | any()) :: Plug.Conn.t()
+  def create(conn, params) do
+    current_user = conn.assigns.current_user
+    content = params["content"]
+
+    case Tweets.create_tweet(%{
+           current_user: current_user,
+           content: content
+         }) do
+      {:ok, _tweet} ->
+        conn
+        |> put_flash(:info, "Post created successfully.")
+        |> redirect(to: Routes.page_path(conn, :index))
+
+      {:error, %Ecto.Changeset{} = _changeset} ->
+        conn
+        |> put_flash(:info, "Invalid Data.")
+        |> redirect(to: Routes.page_path(conn, :index))
+    end
+  end
+
+  @doc """
+  Deletes a post and returns the homepage of the application
+  """
+  @spec delete(Plug.Conn.t(), map() | any()) :: Plug.Conn.t()
+  def delete(conn, %{"id" => tweet_id}) do
+    case Tweets.delete_tweet(tweet_id) do
+      {:ok, _post} ->
+        conn
+        |> put_flash(:info, "Post Deleted Successfully.")
+        |> redirect(to: Routes.page_path(conn, :index))
+
+      {:error, _changeset} ->
+        conn
+        |> put_flash(:info, "Could not delete post.")
+        |> redirect(to: Routes.page_path(conn, :index))
+    end
+  end
+
+  @doc """
+  Updates a post and returns the homepage of the application
+  """
+  @spec update(Plug.Conn.t(), map() | any()) :: Plug.Conn.t()
+  def update(conn, params) do
+    case Tweets.update_tweet(params["id"], %{content: params["content"]["content"]}) do
```

Comment:
```suggestion
  def update(conn, %{"id" => id, "content" => content}) do
    case Tweets.update_tweet(id, %{content: content}) do
```
Use pattern matching. And you shouldn't pass the parameter which nests unnecessarily.